src/test/ui/lint/unused/issue-90807-unused-paren.rs

```diff
@@ -0,0 +1,8 @@
+// check-pass
+// Make sure unused parens lint doesn't emit a false positive.
+// See https://github.com/rust-lang/rust/issues/90807
+#![deny(unused_parens)]
+
+fn main() {
+    for _ in (1..{ 2 }) {}
```

Comment:
Could you also add tests for other kinds of combinations? For example:

```rust
for _ in 1..loop { break 2 } {}
```

```rust
for _ in 1..match () { () => 2 } {}
```

both work and should continue warning, but I think after this PR they won't.

Reply:
Okay, I've tweaked it to do the right thing with the loop and match expressions.